apps/website/src/pages/stream/lola.tsx

```diff
@@ -0,0 +1,91 @@
+import type { InferGetStaticPropsType, NextPage, NextPageContext } from "next";
+import { getSession } from "next-auth/react";
+
+import { env } from "@/env";
+
+import invariant from "@/utils/invariant";
+import { createJWT, signJWT } from "@/utils/jwt";
+
+import LiveCamFeed from "@/components/admin/ptz/LiveCamFeed";
+import Meta from "@/components/content/Meta";
+import { LoginWithTwitchButton } from "@/components/shared/LoginWithTwitchButton";
+
+export async function getServerSideProps(context: NextPageContext) {
+  invariant(env.CF_STREAM_KEY_ID, "CF_STREAM_KEY_ID is not set");
+  invariant(env.CF_STREAM_KEY_JWK, "CF_STREAM_KEY_JWK is not set");
+  invariant(env.CF_STREAM_LOLA_VIDEO_ID, "CF_STREAM_LOLA_VIDEO_ID is not set");
+  invariant(env.CF_STREAM_HOST, "CF_STREAM_HOST is not set");
+
+  const session = await getSession(context);
+  const isAuthed = session?.user?.id !== undefined;
+  const hasRole = session?.user?.roles.includes("ptzControl");
+
+  if (!isAuthed || !hasRole) {
+    return { props: { isAuthed, hasRole } };
+  }
+
+  const expiresTimeInSeconds = 60 * 60 * 48;
+  const expiresIn = Math.floor(Date.now() / 1000) + expiresTimeInSeconds;
+  const token = createJWT(
+    {
+      alg: "RS256",
+      kid: env.CF_STREAM_KEY_ID,
+    },
+    {
+      sub: env.CF_STREAM_LOLA_VIDEO_ID,
+      kid: env.CF_STREAM_KEY_ID,
+      exp: expiresIn,
+      accessRules: [
+        {
+          type: "any",
+          action: "allow",
+        },
+      ],
+    },
+  );
+
+  const jwt = await signJWT(token, env.CF_STREAM_KEY_JWK);
+
+  return {
+    props: {
+      isAuthed,
+      hasRole,
+      url: `https://${env.CF_STREAM_HOST}/${jwt}/webRTC/play`,
+    },
+  };
+}
+
+const PTZPage: NextPage<InferGetStaticPropsType<typeof getServerSideProps>> = ({
```

Comment:
Fundamentally they do the same thing, but I imagine this should use `InferGetServerSidePropsType`?